warehouse/query-core/src/main/java/datawave/query/jexl/JexlASTHelper.java

```diff
@@ -982,6 +982,161 @@ private static Map<String,Object> getAssignments(JexlNode node, Map<String,Objec
+            replaceNodeWithSingleValue(parent, original, replacement);
+        }
+    }
+
+    /**
+     * This method accepts a multivalued replacement node (ASTAndNode or ASTOrNode) and attempts to merge it cleanly into the parent or grandparent node with a
+     * preference for a reference expression as the new parent.
+     *
+     * @param parent
+     *            the parent node
+     * @param original
+     *            the original node
+     * @param replacement
+     *            the replacement node
+     */
+    private static void replaceNodeWithMultiValue(JexlNode parent, JexlNode original, JexlNode replacement) {
+
+        // similar junctions, merge children
+        if (isEquivalentJunctions(parent, replacement)) {
+            mergeJunctions(parent, original, replacement);
+            return;
+        }
+
+        // if no 'legal' rewrite could occur with the parent, try the grandparent
+        JexlNode grandParent = parent.jjtGetParent();
+        if (grandParent != null) {
+            // reference expression, straight replacement
+            if (grandParent instanceof ASTReferenceExpression) {
+                JexlNodes.replaceChild(grandParent, parent, replacement);
+                return;
+            }
+
+            // similar junctions, merge children
+            if (isEquivalentJunctions(grandParent, replacement)) {
+                mergeJunctions(grandParent, parent, replacement);
+                return;
+            }
+        }
+
+        // if no optimal rewrite occurred, fall back to a simple replacement
+        JexlNodes.replaceChild(parent, original, replacement);
+    }
+
+    /**
+     * This method accepts a single replacement node (leaf node) and attempts to merge it cleanly into the parent or grandparent while avoiding a reference
+     * expression as the new parent.
+     *
+     * @param parent
+     *            the parent node
+     * @param original
+     *            the original node
+     * @param replacement
+     *            the replacement node
+     */
+    private static void replaceNodeWithSingleValue(JexlNode parent, JexlNode original, JexlNode replacement) {
+
+        if (!(parent instanceof ASTReferenceExpression)) {
+            // straight replacement
+            JexlNodes.replaceChild(parent, original, replacement);
+            return;
+        }
+
+        // otherwise we have a reference expression and we need to go up a layer
+        JexlNode grandParent = parent.jjtGetParent();
+
+        if (grandParent != null) {
+            // check for the negated single term case: !(term)
+            // it is also unlikely that two reference expressions would sit next to each other in the tree, but it's
+            // not impossible
+            if (!(grandParent instanceof ASTNotNode) && !(grandParent instanceof ASTReferenceExpression)) {
+                JexlNodes.replaceChild(grandParent, parent, replacement);
+                return;
+            }
+        }
```

Comment:
probably also need to explicitly check the type of the parent here?

Reply:
Going back to the calling method, the node being replaced (the parent) can be any node.